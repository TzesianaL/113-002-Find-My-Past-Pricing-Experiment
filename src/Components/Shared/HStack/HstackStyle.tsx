import styled from 'styled-components'

/**
 * Styled components are used in order to style HStack component. Props are passed through ThemeProvider.
 */
export const HStackStyle = styled.div`
  display: flex;
  flex-direction: row;
  font-size: ${(props) => props.theme.fontSize || '1'}rem;
  font-family: ${(props) => props.theme.titleFontFamily};
  width: ${(props) => props.theme.button?.width};
  justify-content: center;
  align-items: center;
`
