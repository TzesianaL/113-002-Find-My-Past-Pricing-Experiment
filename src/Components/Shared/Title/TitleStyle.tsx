import styled from 'styled-components'

/**
 * Styled components are used in order to style Title component. Props are passed through ThemeProvider.
 */
export const TitleStyle = styled.div<{
  subtitle: boolean
}>`
  ${(props) =>
    !props.subtitle &&
    `
    text-align: center;
    text-transform: uppercase;
    font-weight: bold;
    font-family: BoldVf !important;
  `}

  justify-content: center;
  color: ${(props) =>
    props.subtitle
      ? props.theme.subtitleColor || props.theme.colors?.secondary || '#ffffff'
      : props.theme.titleColor || props.theme.colors?.primary};
  font-size: ${(props) =>
    !props.subtitle
      ? props.theme.titleFontSize || props.theme.fontSize || 1.2
      : props.theme.subtitleFontSize || props.theme.fontSize || 1.2}em;
  font-family: ${(props) => props.theme.fontFamily};
  margin: 0.5em 0em 0.5em 0em;
`
