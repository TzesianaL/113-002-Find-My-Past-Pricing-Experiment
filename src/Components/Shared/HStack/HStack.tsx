import React, { FC } from 'react'
import { CSSProperties } from 'styled-components'
import { HStackStyle } from './HstackStyle'

interface HStackProps {
  style?: CSSProperties
}
/**
 * HStack component is used to display horizontally the child component which is passed as prop.
 * HStackStyle is used to style the component
 */
const HStack: FC<HStackProps> = ({ style, children }) => {
  return <HStackStyle style={style}>{children}</HStackStyle>
}

export default HStack
