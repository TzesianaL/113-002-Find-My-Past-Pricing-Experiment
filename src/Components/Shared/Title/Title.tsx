import React, { FC } from 'react'
import { CSSProperties } from 'styled-components'
import { TitleStyle } from './TitleStyle'

export interface TitleProps {
  /**
   * Whether the contents should be displayed using subtitle styling
   */
  subtitle?: boolean
  style?: CSSProperties
}

/**
 * Component that displays any title or subtitle of the tile.
 * TitleStyle is used to style the component
 */
const Title: FC<TitleProps> = ({ style, subtitle, children }) => {
  return (
    <TitleStyle style={style} subtitle={subtitle || false}>
      {children}
    </TitleStyle>
  )
}

export default Title
