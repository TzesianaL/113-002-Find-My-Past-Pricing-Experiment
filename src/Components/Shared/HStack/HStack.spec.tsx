import React from 'react'
import 'jest-styled-components'
import * as renderer from 'react-test-renderer'
import HStack from './HStack'
import Title from '../Title/Title'

describe('<HStack/>', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <HStack>
          <Title>Test Title</Title>
        </HStack>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
