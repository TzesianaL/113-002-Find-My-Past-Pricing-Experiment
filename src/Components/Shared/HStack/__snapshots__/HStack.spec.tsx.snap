// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HStack/> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
  font-family: BoldVf !important;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.2em;
  margin: 0.5em 0em 0.5em 0em;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Test Title
  </div>
</div>
`;
